### PR TITLE
Add deterministic AI evaluation policy with score normalization

### LINKED ISSUE #34: POC: Add Deterministic AI Evaluation Policy and Normalization Rules
## Summary
Add a deterministic evaluation and normalization policy for AI work so token rewards are reproducible instead of arbitrary.

## Problem to solve
An AI work token system needs a rule for turning work output into economic value. If scoring is vague, hidden, or non-deterministic, then reward issuance cannot be trusted or tested. The repository currently has no business logic yet, so this policy should be introduced early as a reusable foundation. 3

## Proposed solution
Create an evaluation policy module that:
- accepts a task and submission
- produces a raw score
- normalizes that raw score into an explicitly allowed discrete score set
- records evaluator version and normalization policy metadata

Requirements:
- normalization must be deterministic and documented
- no “nearest guess” wording; use an explicit policy
- the same input and submission under the same evaluator version must always produce the same normalized score
- implementation should prefer minimal working rules over speculative AI/LLM-heavy scoring

Suggested direction:
- start with rule-based or fixture-based scoring for the POC
- place policy logic in a reusable module/package that can later be consumed by services
- add unit tests covering score boundaries and normalization behavior

## Acceptance criteria
- Deterministic score normalization function exists
- Allowed score set is explicitly defined in code
- Policy documentation explains how raw score becomes normalized score
- Unit tests prove stable output for repeated identical inputs

## Impact
This establishes the trust layer for the project: AI-defined value becomes reproducible and testable rather than hand-wavy.

### PR DESCRIPTION
**Pull request by claude-bot-derinworks**
## Summary
Introduces a new evaluation policy module that provides deterministic normalization of raw AI work scores into a discrete set of allowed scores (0.0 to 1.0 in 0.1 increments). This ensures consistent, reproducible evaluation results across the system.

## Key Changes
- **New module**: `shared/evaluation_policy.py` containing:
  - `normalize()` function that converts raw scores to discrete allowed scores via clamping, scaling, and rounding
  - `evaluate()` high-level helper that wraps normalization and returns an `EvaluationResult` with metadata
  - `EvaluationResult` Pydantic model for immutable evaluation records with validation constraints
  - Policy constants: `POLICY_VERSION`, `ALLOWED_SCORES` (11 discrete levels), and `DEFAULT_RAW_SCORE_MAX`

- **Comprehensive test suite**: `tests/test_evaluation_policy.py` with 160 lines covering:
  - Allowed scores validation (11 levels from 0.0 to 1.0)
  - Normalization behavior (clamping, scaling, banker's rounding to nearest tenth)
  - Custom raw score max support
  - Determinism guarantees across repeated calls
  - Pydantic model constraints and serialization roundtrips
  - Policy version metadata

- **Module export**: Updated `shared/__init__.py` to expose the new `evaluation_policy` module

## Notable Implementation Details
- **Determinism guarantee**: Identical inputs always produce identical outputs with no randomness or floating-point tolerance heuristics
- **Banker's rounding**: Uses Python's `round()` for rounding to nearest tenth (e.g., 0.75 → 0.8, 0.85 → 0.8)
- **Validation**: `EvaluationResult` enforces normalized_score ∈ [0.0, 1.0] and raw_score_max > 0 via Pydantic Field constraints
- **Semantic versioning**: Policy version must be bumped for any changes to normalization logic

https://claude.ai/code/session_01J3YtPdRhhaz5uK4RhJhEFB

Closes #34

### CODE REVIEW

File: shared/evaluation_policy.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# ---------------------------------------------------------------------------
+
+
+class EvaluationResult(BaseModel):
+    """Immutable record produced by :func:`evaluate`."""
+
+    raw_score: float
+    normalized_score: float = Field(ge=0.0, le=1.0)
+    raw_score_max: float = Field(gt=0)
+    policy_version: str
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply shared/evaluation_policy.py]: Out of scope for this PR. `EvaluationResult` is the output of the normalization policy (raw→discrete conversion with version metadata), while `AIWorkEvaluation` is a pipeline record with IDs and addresses. They serve different purposes. Tracked separately.

**Comment by codex-bot-derinworks** [comment]
@codex

**Comment by codex-bot-derinworks** [comment]
@codex

**Comment by codex-bot-derinworks** [comment]
@codex

File: shared/evaluation_policy.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    # 1. Clamp
+    clamped = max(0.0, min(raw_score, raw_score_max))
+
+    # 2. Scale to [0, 1]
+    continuous = clamped / raw_score_max
+
+    # 3. Discretize – round half-up to nearest tenth via floor(x + 0.5)
+    #    to avoid Banker's Rounding behaviour of Python's built-in round().
+    nearest_index = floor(continuous * 10 + 0.5)
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply shared/evaluation_policy.py:109]: Out of scope for this PR. This was already addressed in the previous round — the code and comments correctly describe round-half-up behaviour. The module docstring says "rounding to the closest discrete level" which is accurate and strategy-agnostic.

File: shared/evaluation_policy.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    raw_score_max: float = Field(gt=0)
+    policy_version: str = POLICY_VERSION
+
+    @field_validator("normalized_score")
+    @classmethod
+    def must_be_allowed_score(cls, v: float) -> float:
+        if v not in ALLOWED_SCORES:
+            raise ValueError(
+                f"normalized_score must be one of {ALLOWED_SCORES}, got {v}"
+            )
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply shared/evaluation_policy.py:132]: Out of scope for this PR. The issue requires "explicit policy rules (avoid vague approximations)" and deterministic normalization. Using tolerance-based comparison (isclose) would undermine that guarantee. Values produced by normalize() come directly from the ALLOWED_SCORES tuple, and JSON deserialization preserves exact single-decimal floats. Tracked separately.

**Review by admin-derinworks — APPROVED** [review]
